python/paddle/incubate/autograd/composite_rules.py

```diff
@@ -411,6 +411,30 @@ def silu_composite(x):
     return res
 
 
+@REGISTER_COMPOSITE('meshgrid')
+def meshgrid_composite(inputs):
+    """
+    define composite rule of op meshgrid
+    If the input has N tensors of size S_0, ... S_n-1, then the output will also have N tensors, where
+    each tensor is of shape (S_0, ..., S_n-1)
+    """
+    size = len(inputs)
+    shape = [1 for i in range(size)]
+    for i in range(size):
+        dim = inputs[i].dim()
+        assert dim == 0 or dim == 1
+        if dim == 0:
+            shape[i] = 1
+        else:
+            shape[i] = inputs[i].shape[0]
```

Comment:
may the logic be simplified?

Reply:
Yes, this op can support zero dim tensor, so we should consider 0 dim tensor as shape [1]